tests/suites/casper/wait.js

```diff
@@ -167,3 +167,149 @@ casper.test.begin('waitForUrl() string tests', 1, function(test) {
+                test.assertEquals(details.data.exitCode, null, 'Non existing executable returned exitCode null');
+                test.assertEquals(details.data.elapsedTime, null, 'Non existing executable returned elapsedTime null');
+        });
+
+        if (system.os.name != "windows") {
+            if (lsExecutable) {
+
+                casper.reload().waitForExec(lsExecutable, ['-a',fs.workingDirectory],
+                    function success3(details) {
+                        test.assert((utils.isString(details.data.command) && utils.isArray(details.data.parameters)), 'Casper.waitForExec() can call "lsExecutable" on "workingDirectory": ' + JSON.stringify(details.data));
+                        test.assert( (utils.isNumber(details.data.pid) && ( details.data.pid > 0) ), '"lsExecutable" on "workingDirectory" used PID ' + details.data.pid);
+                        test.assertEquals(details.data.isChildNotFinished, 0,'"lsExecutable" on "workingDirectory" returned "isChildNotFinished" 0');
+                        test.assertEquals(details.data.exitCode, 0, '"lsExecutable" on "workingDirectory" returned exitCode 0');
+                        test.assertTruthy(details.data.stdout, '"lsExecutable" on "workingDirectory" returned stdout ' + JSON.stringify(details.data.stdout));
+                });
+
+                casper.reload().waitForExec(lsExecutable, [notExecutable],
+                    function success4(details) {
+                        test.assert((utils.isString(details.data.command) && utils.isArray(details.data.parameters)), 'Casper.waitForExec() can call "lsExecutable" on "notExecutable": ' + JSON.stringify(details.data));
+                        test.assert( (utils.isNumber(details.data.pid) && ( details.data.pid > 0) ), '"lsExecutable" on "notExecutable" used PID ' + details.data.pid);
+                        test.assertEquals(details.data.isChildNotFinished, 0,'"lsExecutable" on "notExecutable" returned "isChildNotFinished" 0');
+                        test.assertTruthy(details.data.exitCode, '"lsExecutable" on "notExecutable" returned exitCode ' + details.data.exitCode);
+                        test.assertTruthy(details.data.stderr, '"lsExecutable" on "notExecutable" returned stderr ' + JSON.stringify(details.data.stderr));
+                });
+
+            } else {
+                casper.reload().then( function() {
+                    test.skip(10, 'ls EXECUTABLE NOT FOUND');  
+                });
+            };
+
+            if (shExecutable) {
+                // vars used to measure if 'shell script to trap TERM signal' is waiting enough and being killed faster enough
+                var shellTrapTimeout = 500;
+                var shellTrapTermTimeout = 1000;
+                var timeLapseAfterKillSignal = 300;
+                var minTimeLapse = shellTrapTimeout + shellTrapTermTimeout;
+                var maxTimeLapse = minTimeLapse + timeLapseAfterKillSignal;
+                casper.then( function() {
```

Comment:
Must be inside a then. Otherwise the elapsedTime will be the sum of other tests an this test.

With then, elapsedTime is something like 1600, 1700ms

Without then it's is something like 4000ms (remember first test usually takes 2000, this 1600, so we have 3600ms plus the other tests.